folly/Range.h

```diff
@@ -474,7 +474,7 @@ class Range {
   template <typename ValueType>
   struct StringViewType
       : std::conditional<
-            std::is_pod<std::remove_const_t<ValueType>>::value,
+            folly::is_trivially_copyable<std::remove_const_t<ValueType>>::value,
```

Comment:
This change has had a chance to run on internal CI. There, it is observed to cause a large number of build failures.

In practice, libstdc++ `std::basic_string_view` has a static-assert that the type parameter is trivial. We use this test to choose whether conversion to `std::basic_string_view` is available, so if the condition here is too broad, the test can allow conversion to an instance of `std::basic_string_view` which immediately fails a static-assert.

Can we switch this to `std::is_trivial`?

Reply:
thanks for the heads up; yeah let's try is_trivial, possibly combined with is_standard_layout; 
I think the original trait was probably not right either as it allows the elements in the string view to be arrays which seems wrong

```suggestion
            std::is_trivial<std::remove_const_t<ValueType>>::value,
```